code/datums/components/plumbing/reaction_chamber.dm

```diff
@@ -9,26 +9,34 @@
 
 /datum/component/plumbing/reaction_chamber/can_give(amount, reagent, datum/ductnet/net)
 	. = ..()
-	var/obj/machinery/plumbing/reaction_chamber/RC = parent
-	if(!. || !RC.emptying || reagents.is_reacting == TRUE)
+	var/obj/machinery/plumbing/reaction_chamber/reaction_chamber = parent
+	if(!. || !reaction_chamber.emptying || reagents.is_reacting == TRUE)
 		return FALSE
 
 /datum/component/plumbing/reaction_chamber/send_request(dir)
-	var/obj/machinery/plumbing/reaction_chamber/RC = parent
-	if(RC.emptying)
+	var/obj/machinery/plumbing/reaction_chamber/chamber = parent
+	if(chamber.emptying)
 		return
 
-	process_request(amount = min(MACHINE_REAGENT_TRANSFER, RC.target_volume - reagents.total_volume), dir = dir)
-
-	if(RC.target_volume > round(reagents.total_volume, CHEMICAL_VOLUME_ROUNDING)) //not enough yet
-		return
+	for(var/required_reagent in chamber.required_reagents)
+		var/has_reagent = FALSE
+		for(var/containing_reagent_but_not_typecasted_yet in reagents.reagent_list)
+			var/datum/reagent/containg_reagent = containing_reagent_but_not_typecasted_yet as anything
```

Comment:
Not how it works--here's the page on it: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists